### DISCUSSION TITLE
Update packages based on a dist tags

### DISCUSSION THREAD

**Discussion by wehrstedt** [renovatebot/renovate#9763] [Request Help] [answered]
Hello everyone. I'm trying to achive the following:
* We have a private hosted verdaccio registry for our internal packages
* we have a dist tag named "develop" where beta-releases of our packages are published
* i'd like renovate to update to the latest available version of that dist tag

To make it more clear, lets assume the following example:

**module-A** is published with the following versions in our verdaccio registry:
Dist tag | Version
------------ | -------------
latest | 1.1.0
develop | 1.2.0-develop-233gc2fbb3f

With the following version history
Version | Date
------------ | -------------
1.2.0-develop-233gc2fbb3f | 43 minutes ago
1.2.0-develop-221ab3s2df | 2 days ago
1.1.0 | 1 month ago

In my *main project* i installed *module-A* like this:
_package.json_
```json
"dependencies": {
  "module-A": "^1.1.0"
}
```

Now i'd like renovate to automatically update always to the latest published version by dist-tag "develop", which is currently `1.2.0-develop-233gc2fbb3f`. To achive this, I updated the renovate config if my project as described [here](https://github.com/renovatebot/config-help/issues/18) like this:
_renovate.json_
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "renovate/renovate-config" // this is my self hosted project with a default config
  ],
  "respectLatest": false,            // this entry was provided via vscode autocomplete
  "packageRules": [                  // this is the solution from https://github.com/renovatebot/config-help/issues/18
    {
      "enabled": true,
      "matchPackageNames": [
        "module-A"
      ],
      "respectLatest": false
    }
  ]
}
```

I would now expect renovate to update `module-A` to version `1.2.0-develop-233gc2fbb3f`, but nothing happened.
I also tried `"matchDatasources": [ "npm" ]`, with and without `enabled` and different combinations of `respectLatest`.

Is the described behavoir even possible with renovate? Thanks for your help!

**Comment by rarkins** [comment, marked as answer]
The config option `followTag` should satisfy this. As in `followTag=develop`. Using regular semver sorting won't work because git hashes don't sort in calendar order

**Reply by wehrstedt** [reply]
Yes, thank you! With this option it finally works, but with one side effect: Now renovate performs a rollback if the dist tag is behind the latest tag. Can this option be combined with `respectLatest=true`?

**Reply by rarkins** [reply]
I'm not sure how those two interact. Rollback should only occur if the current constraint doesn't match any published versions on npm (eg version you're using gets removed from the registry). You can disable rollback PRs through packageRules also

**Reply by wehrstedt** [reply]
In my case, i installed *module-B* with version `^0.13.0`:
_package.json_
```json
"devDependencies": {
		"module-B": "^0.13.0"
}
```

Then, I added `followTag=develop`
_renovate.json_
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "renovate/renovate-config"
  ],
  "followTag": "develop"
}
```

Then, renovate sends me this MR:
| Package | Type | Update | Change |
|---|---|---|---|
| module-B | devDependencies | rollback | `^0.13.0` -> `^0.2.0` |

And these are the published versions:
| Dist-Tag | Version |
|---|---|
| latest | 0.13.2 |
| develop | 0..2.0 |

**Reply by rarkins** [reply]
But that *is* a rollback/downgrade, isn't it?

**Reply by wehrstedt** [reply]
Hmm .. Good question. Personaly, in that case I want to keep 0.13.2, because `0.13.2 > 0.2.0`. I can imagine that someone else sees it differently.

Well, the above scenario is an extreme example. Maybe this example will make it more clear what i am trying to achive:
* we publish multiple beta-releases, lets say `0.13.3-10geslciq` and `0.13.3-12kdlekdka`
* some day, we decide to make a stable release. We remove the hash and version `0.13.3` will be published with tag latest.
* Now I have 0.13.3-12kdlekdka -> develop and 0.13.3 -> latest
* What I want now is that renovate installs `0.13.3`, because it's the newer version (just regarding the timestamp)
* As time goes on, we continue develop and publish new beta-releases, like `0.13.4-2laiekdkv`
* Now `0.13.4-2laiekdkv > 0.13.3` --> renovate should install `0.13.4-2laiekdkv`

That ensures that I will always have the latest version installed in multiple projects and i will notice pretty fast if i broke something, have always nightly builds with the latest possible modules installed etc.

While writing down my usecase I noticed that i can achive the behaviour by re-tag develop to the same version as latest, once i draft a release which will become latest. So instead of
> Now I have 0.13.3-12kdlekdka -> develop and 0.13.3 -> latest

I would have 0.13.3 -> develop and 0.13.3 -> latest.

And maybe our way of working is totally unusual and nobody undestands why we are working like that 😄

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#ignoreunstable

So you don't like to follow a dist tag. You like to always have newest version including unstable. So follow docs above

**Reply by wehrstedt** [reply]
I added this option now. It seems like this is exactly what i want. I will give it a try, currently our registry is a little bit messed up because of testing. I will report back. Thank you!
